metsim/metsim.py

```diff
@@ -209,169 +226,193 @@ def _validate_force_times(self, force_times):
                                       pd.Timedelta("90 days"))
         self.params['state_stop'] = (self.params['start'] -
                                      pd.Timedelta("1 days"))
-        logger.info('start {}'.format(self.params['start']))
-        logger.info('stop {}'.format(self.params['stop']))
+        print('start {}'.format(self.params['start']))
+        print('stop {}'.format(self.params['stop']))
 
-        logger.info('force start {}'.format(pd.Timestamp(
+        print('force start {}'.format(pd.Timestamp(
             force_times.values[0]).to_pydatetime()))
-        logger.info('force stop {}'.format(pd.Timestamp(
+        print('force stop {}'.format(pd.Timestamp(
             force_times.values[-1]).to_pydatetime()))
 
-        logger.info('state start {}'.format(self.params['state_start']))
-        logger.info('state stop {}'.format(self.params['state_stop']))
+        print('state start {}'.format(self.params['state_start']))
+        print('state stop {}'.format(self.params['state_stop']))
 
-        logger.info('calendar {}'.format(self.params['calendar']))
+        print('calendar {}'.format(self.params['calendar']))
         if self.params['utc_offset']:
-            attrs['time'] = {'long_name': 'UTC time',
+            attrs['time'] = {'units': DEFAULT_TIME_UNITS,
+                             'long_name': 'UTC time',
                              'standard_name': 'utc_time'}
         else:
-            attrs['time'] = {'long_name': 'local time at grid location',
+            attrs['time'] = {'units': DEFAULT_TIME_UNITS,
+                             'long_name': 'local time at grid location',
                              'standard_name': 'local_time'}
 
+    @property
+    def domain(self):
+        if self._domain is None:
+            print("read_domain")
+            self._domain = io.read_domain(self.params).isel(
+                **self._domain_slice)
+        return self._domain
+
+    @property
+    def met_data(self):
+        if self._met_data is None:
+            print("read_met_data")
+            self._met_data = io.read_met_data(self.params, self.domain).isel(
+                **self._domain_slice)
+            self._met_data['elev'] = self.domain['elev']
+            self._met_data['lat'] = self.domain['lat']
+            self._validate_force_times(force_times=self._met_data['time'])
+        return self._met_data
```

Comment:
we need to clip to the shape of the domain here (like we currently do in read_netcdf)